tools/filter_by_fasta_ids/filter_by_fasta_ids.py

```diff
@@ -64,14 +63,21 @@ def main():
     parser.add_argument('-d', help='Path to discarded entries file')
     header_criteria = parser.add_mutually_exclusive_group()
     header_criteria.add_argument('--id_list', help='Path to the ID list file')
+    parser.add_argument('--pattern', default=None, help='regex earch attern for ID in Fasta entry')
     header_criteria.add_argument('--header_regexp', help='Regular expression pattern the header should match')
     sequence_criteria = parser.add_mutually_exclusive_group()
     sequence_criteria.add_argument('--min_length', type=int, help='Minimum sequence length')
     sequence_criteria.add_argument('--sequence_regexp', help='Regular expression pattern the header should match')
     parser.add_argument('--max_length', type=int, help='Maximum sequence length')
     parser.add_argument('--dedup', action='store_true', default=False, help='Whether to remove duplicate sequences')
     options = parser.parse_args()
-
+    
+    
+    pattern =  options.pattern if options.pattern else '>([^| ]+)'
+    if not re.match('^.*[(](?![?]:).*[)].*$',pattern):
+        print('pattern: "%s" did not include "%%s"' % pattern)
```

Comment:
is %% really want you want?

Reply:
Yes.

Reply:
We want user to specify ( ) in the regex for capturing group which can be used to pass in the string to match the fasta header.

Reply:
I mean in the string substitution :)

Reply:
How about if I change it to 

print('pattern: "%s" did not include capture group "()" in regex ' % pattern)

Reply:
sure

Reply:
Done!! Thanks a lot!!